### PR TITLE
#127 - Fix hasData return while data is false when casted as a boolean

### PR DESCRIPTION
**Pull request by mcsky**
Like @khylias figured out the `hasData` function from the ModalComponent works badly.
Unfortunately the function return `false` with some others values than `0`  please see [this fiddle](https://jsfiddle.net/mfua5rqd/1/)

This PR fix the issue

### CODE REVIEW

**Comment by mcsky** [comment]
Actually, I learned yesterday that `null` could be considered as a value.
By default, if you define a property without value in a class it is initialized to `undefined`.
IMO we should test only `!== undefined`, what do you think?

**Comment by maximelafarie** [comment]
I only saw your last comment now. Yep, `undefined` is the essential check to perform. Let's do this! 👌

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
   public overlayVisible: boolean = false;
   public openedClass: boolean = false;
 
-  private _data: any = null;
+  private _data: any; // No value => undefined
```
Comment (maximelafarie) [inline src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: Remove this comment! 😂

**Comment by khylias** [comment]
What about this issue ? https://github.com/biig-io/ngx-smart-modal/issues/129

And it seems to that we can close this issue 😄 [#127](https://github.com/biig-io/ngx-smart-modal/issues/127)

**Pull request merged**